run

```diff
@@ -885,38 +893,50 @@ for k, v in obj[project_type]["run"].items():
 #===========================================================================================
 # Other options
 
+list_cmake_dir_options() {
+  cmake_function=$1
+  cmakelists=$(find ${SCRIPT_DIR} -name "CMakeLists.txt")
+  cmake_add_dir=$(grep --no-filename $cmake_function $cmakelists)
+  echo "$cmake_add_dir" | sed -n "s|$cmake_function([[:space:]]*\([^[:space:])]*\).*|\1|p" | sort
+}
+
+list_apps() {
+  list_cmake_dir_options add_holohub_application
+}
+
+list_operators() {
+  list_cmake_dir_options add_holohub_operator
+}
+
+list_packages() {
+  list_cmake_dir_options add_holohub_package
+}
+
 # Build HoloHub sample apps
 list_desc() {
   echo ""
-  echo "Display the list of applications to run."
+  echo "Display the list of packages, applications, and operators to build."
   echo "Usage: ./run list"
   echo ""
 }
 
 list() {
```

Comment:
looks like the `autocompletion_list` [below](https://github.com/nvidia-holoscan/holohub/pull/639/files#diff-acba25512100f80b56fc3ccd14c65be55d94800cda77585c5f41a887e398f9beR943) could also be updated to use some of the logic changed here... though it might need more love than that, since I was confused why apps where even listed, or what some of the options in there related to (cpp, python).

Happy to update it if I get some confirmation on the expected behavior and on the parts that don't currently seem correct in there.

Reply:
@tbirdso thoughts?

Reply:
Going to defer to @jjomier as I haven't touched the autocomplete implementation before. I'm OK with moving forward without in this PR and updating as necessary.